sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2911,4 +2911,28 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitRefreshTable(ctx: RefreshTableContext): LogicalPlan = withOrigin(ctx) {
     RefreshTableStatement(visitMultipartIdentifier(ctx.multipartIdentifier()))
   }
+
+  /**
+   * A command for users to list the column names for a table.
+   * This function creates a [[ShowColumnsStatement]] logical plan.
+   *
+   * The syntax of using this command in SQL is:
+   * {{{
+   *   SHOW COLUMNS (FROM | IN) tableName=multipartIdentifier
+   *        ((FROM | IN) namespace=multipartIdentifier)?
+   * }}}
+   */
+  override def visitShowColumns(ctx: ShowColumnsContext): LogicalPlan = withOrigin(ctx) {
+    import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+
+    val table = visitMultipartIdentifier(ctx.table)
+    val namespace = Option(ctx.namespace).map(visitMultipartIdentifier)
+    if (namespace.isDefined && namespace.get.length > 1 && table.length > 1) {
+      val tableName = table.init
+      throw new ParseException(
+        s"If namespace is multipart, table can not be multipart: ${tableName.quoted}", ctx)
```

Comment:
Just table.quoted.  visitCacheTable takes tableName.init because it only wants catalog and namespace there.

In your case, you need to show full given multipart table.

Reply:
how about "Cannot specify multi-part table name if `FROM namespace` is specified."

Reply:
If the behavior is as https://github.com/apache/spark/pull/26182#discussion_r339892210, then the suggested one is more correct.